Preview different Plan UI's with a radio toggle

dereksmart opened the pull request:
This PR adds a new component `DevCard` to the mix, which is rendered in the footer only if Jetpack is running in a **development version**.  

Right now it only allows you to toggle between different Jetpack plans so that you're able to view the **UI** differences.  

_**Toggling between these is not actually changing your plan, it is only updating the state for viewing/testing purposes of the UI**_ -- aka: Your plan is expected to revert back to your actual plan when the /site endpoint is queried next, so that is NOT a bug.  

There is no doubt that this can be improved/expanded a great deal in the future (and it will), but for now **this PR is meant to be a quick win for designers/growth when they are testing differences between plans**. 

**To Test:** 
- Checkout and view the `/wp-admin/admin.php?page=jetpack#/plans` tab.  
- Toggle between the plans.  The view should reflect accordingly.  
- Navigate to the dashboard.  Your plan should be updated along with the /site endpoint response.  
- Change the Jetpack version in jetpack.php to `4.8`, make sure the component does not render.  

![switchplan](https://cloud.githubusercontent.com/assets/7129409/23148130/cfc86fca-f7b1-11e6-8dc1-ae510d566553.gif)

samhotchkiss commented on _inc/client/components/dev-card/index.jsx:
```diff
@@ -0,0 +1,106 @@
+ * Internal dependencies
+ */
+import { isDevVersion as _isDevVersion } from 'state/initial-state';
+import { switchMichaelsPlan } from 'state/dev-version';
```
Let's not use people's names in our codebase?

samhotchkiss commented on _inc/client/state/dev-version/actions.js:
```diff
@@ -35,4 +36,13 @@ export const resetOptions = ( options ) => {
-}
+};
+
+export const switchMichaelsPlan = ( slug ) => {
```
How about "previewAnotherPlan" or something like that?

samhotchkiss commented:
Noice!

Should we just be showing this to all users running a dev version?  Our goal over the next few months is to get many more beta testers, and this seems like it might be a bit unnecessary/confusing.  What about a little footer link to turn this thing on?  Or...

dereksmart commented:
@samhotchkiss I was thinking of putting it behind a filter

dereksmart commented:
I've updated this to add state/actions to be able to toggle the devcard.  I've added a link to toggle it specifically in the footer in dev versions only, similar to how we have `reset options`.  

This is what it looks like 
![switchplan-toggle](https://cloud.githubusercontent.com/assets/7129409/23169670/afd19ab6-f81a-11e6-9dcb-aa1aaa9fea12.gif)

samhotchkiss commented:
Love it!
